### ISSUE TITLE
Validate Params in Base Class

### ISSUE THREAD

**Issue by dilyabareeva** [dilyabareeva/quanda#17]
Problem: We want to validate arguments passed to metrics. Validating in every metric separately creates a lot of code and makes the metric code less readable. 

I suggest we create a class attribute parameter_constraints dictionary for every metric, which includes parameters:
- name
- constraint type
- constraint values. 

Then we create a centralized validate_param method in base metric or decorator to iterate through every param and validate it.

Typical parameter types:
- python primitives
- our classes
- int/float interval
- string options

Inspired by scikit-learn: https://github.com/scikit-learn/scikit-learn/blob/d8ee3fcb49625362e8206529466ae5b9a3b83dad/sklearn/base.py
